lib/ddtrace/configuration/validation_proxy.rb

```diff
@@ -0,0 +1,44 @@
+require 'forwardable'
+
+module Datadog
+  module Configuration
+    # Forwards configuration settings allowed for tracing,
+    # but raises errors for access to anything else.
+    class TracingValidationProxy
+      extend Forwardable
+
+      FORWARDED_METHODS = [
+        :analytics,
+        :api_key,
```

Comment:
I believe `api_key` is not used by tracing (it's only used for profiling)

Reply:
Yes, this is true, but we may want to use it in tracing or other applications.

In general, I think it's okay if we allow access to global configuration values through the specific namespace, as they will be accessible to those features anyways. I suggest we leave this behavior as written until we can do a more thorough rework of configuration in 1.x, as this is very usable and doesn't box us in on any decisions regarding configuration.

Reply:
We talked more about this. We decided we will put global vars behind `Datadog.configure` and tracing vars behind `Datadog::Tracing.configure` with validation proxies on each.